python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -82,16 +87,47 @@ export default function EditorContainer({
     }
   }, [aiconfigState, callbacks.save]);
 
+  const debouncedUpdatePrompt = useMemo(
```

Comment:
Not 100% confident this debouncing should be done here vs in the Editor.tsx; I did it here since it seems like it should be 'our' responsibility to debounce external update calls instead of having to implement debounce in Editor.tsx and in the gradio root component